File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <math.h>
#include <string.h>



bool is_armstrong_number(int candidate);



bool is_armstrong_number(int candidate){

    int soma;
    int digit;
    int counter = 0;

    char numberString[10];
    itoa(candidate, numberString, 10);

    for(int i=0; i < strlen(numberString); i++){
        if(isdigit(numberString[i])){
            counter ++;
        }
    }

    for(int i=0; i < strlen(numberString); i++){
        digit = numberString[i] - '0';

        soma += (int)round(pow(digit, counter)); 
    }

    if(candidate == soma){
        return true;
    }

    return false;

}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {

        String numberString = String.valueOf(numberToCheck);
        int result = 0;
        for(int i = 0; i < numberString.length(); i++) {
            char digitChar = numberString.charAt(i);
            int digit = Character.getNumericValue(digitChar);

            result += Math.pow(digit, numberString.length());
        }

        return result == numberToCheck;

    }

}
